src/Microsoft.AspNetCore.Localization/CookieRequestCultureProvider.cs

```diff
@@ -36,6 +42,8 @@ public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpC
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
+            _logger = httpContext.RequestServices.GetService<ILogger>() ?? NullLogger.Instance;
```

Comment:
BTW I want to make sure there are no breaking changes here, so be careful about adding a ctor if there wasn't one before. If we add a new parameterized ctor, we'd need to add a new empty ctor as well for compat (and make sure that things actually work with both ctors).